python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -525,6 +526,21 @@ def _get_mirror_list(self, repo, url):
         mirrorlist_path = os.path.join(repo.root, 'mirrorlist.txt')
         returnlist = []
         content = []
+        # If page not plaintext or xml, is not a valid mirrorlist or metalink,
+        # so continue without it.
+        proxies = get_proxies(self.proxy_url, self.proxy_user, self.proxy_pass)
+        cert = (self.sslclientcert, self.sslclientkey)
+        verify = self.sslcacert
+        try:
+            webpage = requests.get(url, proxies=proxies, cert=cert, verify=verify)
+            content_type = webpage.headers["Content-Type"]
+            if "text/plain" not in content_type and "xml" not in content_type:
+                # Not a valid mirrorlist or metalink; continue without it
+                return returnlist
+        except Exception as exc:
```

Comment:
I'm not sure if the preferred course is to remove it entirely so it fails, or to handle each exception specifically; if it's the latter then I'd rather do that in a followup patch because it may take some time to work out what needs to be there.

Reply:
It seems all exceptions raised by `requests` are inherited from `requests.exceptions.RequestException`, so we could use that one.